src/overlays/actors/ovl_En_Dinofos/z_en_dinofos.c

```diff
@@ -114,144 +257,1279 @@ static DamageTable D_8089E314 = {
+            return 1;
+        }
+
+        if (this->actor.colChkInfo.damageEffect == 2) {
+            this->unk_2B0 = 4.0f;
+            this->unk_2B4 = 0.55f;
+            this->unk_28A = 0;
+        } else if (this->actor.colChkInfo.damageEffect == 4) {
+            this->unk_2B0 = 4.0f;
+            this->unk_2B4 = 0.55f;
+            this->unk_28A = 20;
+            Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_CLEAR_TAG,
+                        this->colliderJntSph.elements[i].info.bumper.hitPos.x,
+                        this->colliderJntSph.elements[i].info.bumper.hitPos.y,
+                        this->colliderJntSph.elements[i].info.bumper.hitPos.z, 0, 0, 0, CLEAR_TAG_LARGE_LIGHT_RAYS);
+        }
+        func_8089C87C(this, i);
+        return 1;
+    }
+
+    return 0;
+}
+
+void EnDinofos_Update(Actor* thisx, GlobalContext* globalCtx2) {
+    EnDinofos* this = THIS;
+    GlobalContext* globalCtx = globalCtx2;
+    s32 pad;
+    Vec3f sp30;
+
+    if (this->actionFunc != func_8089C7B8) {
+        func_8089A900(this);
+    }
+
+    if (!func_8089D60C(this, globalCtx) && (this->colliderQuad.base.atFlags & 4)) {
+        func_8089C690(this);
+    }
+
+    this->actionFunc(this, globalCtx);
+    Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 25.0f, 30.0f, 60.0f, 0x5D);
+    if (this->actionFunc != func_8089C7B8) {
+        if ((this->actor.yDistToWater > 0.0f) && (this->actor.yDistToWater < 10.0f)) {
+            if (!((globalCtx->gameplayFrames & 3) & 1)) {
+                Math_Vec3f_Copy(&sp30, &this->unk_2D4[1 + (globalCtx->gameplayFrames & 3)]);
+                sp30.y = this->actor.world.pos.y + this->actor.yDistToWater;
+                EffectSsGRipple_Spawn(globalCtx, &sp30, 0x64, 0x140, 0);
+            }
+        }
+        if (this->actionFunc != func_8089D018) {
+            func_8089D42C(this, globalCtx);
+        }
+    }
+
+    if ((this->actionFunc == func_8089D018) || (this->actionFunc == func_8089C938)) {
+        Math_ScaledStepToS(&this->unk_28E, 0, 0x7D0);
+    }
+
+    Math_Vec3f_Copy(&this->actor.focus.pos, &this->unk_304);
+    this->actor.focus.rot.y = this->actor.shape.rot.y + this->unk_28E;
+    if (this->colliderJntSph.base.acFlags & 1) {
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->colliderJntSph.base);
+    }
+
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->colliderJntSph.base);
+    if (this->colliderJntSph.base.atFlags & 1) {
+        CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->colliderJntSph.base);
+    }
+
+    if (this->colliderQuad.base.atFlags & 1) {
+        CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->colliderQuad.base);
+    }
+
+    if (this->unk_2B0 > 0.0f) {
+        if (this->unk_28A != 10) {
+            Math_StepToF(&this->unk_2B0, 0.0f, 0.05f);
+            this->unk_2B4 = (this->unk_2B0 + 1.0f) * 0.275f;
```

Comment:
```suggestion
            this->unk_2B4 = (this->unk_2B0 + 1.0f) * (11.0f / 40.0f);
```

Reply:
Is that really preferable? Where are those numbers coming from? What does 11 and 40 represent?

Reply:
I tend to try and have things in fractions if it's not obvious what fraction it is to begin with, just a personal preference. It's hard to tell what 11 and 40 represents without documentation, but most of the fractions in this block of code look to deal with ratio's of `20` or `40` or whatnot, like `11.0f / 20.0f` (0.05), `11.0f / 40.0f` (0.275), `11.0f / 800.0f` (0.01375f), all seem related